arcade/paths.py

```diff
@@ -266,19 +341,23 @@ def astar_calculate_path(start_point: Point,
 def has_line_of_sight(point_1: Point,
                       point_2: Point,
                       walls: SpriteList,
-                      max_distance: int = -1,
+                      max_distance: float = float("inf"),
                       check_resolution: int = 2) -> bool:
     """
-    Determine if we have line of sight between two points. Try to make sure
-    that spatial hashing is enabled on the wall SpriteList or this will be
-    very slow.
-
-    :param Point point_1: Start position
-    :param Point point_2: End position position
-    :param SpriteList walls: List of all blocking sprites
-    :param int max_distance: Max distance point 1 can see
-    :param int check_resolution: Check every x pixels for a sprite. Trade-off
-                                 between accuracy and speed.
+    Determine if we have line of sight between two points.
+
+    NOTE: Try to make sure that spatial hashing is enabled
+    on the wall SpriteList or this will be very slow.
+    NOTE: Make sure max_distance is positive.
```

Comment:
I would remove this  line and add an if statement bounds check at the top of the function:
```py
    if max_distance <= 0:
        raise ValueError("max_distance must be greater than zero")
```

Reply:
Yeah, I don't think we need that.
`
    if max_distance <= 0:
        raise ValueError("max_distance must be greater than zero")
`

If it is negative, it will return false no matter what. How about a waring instead?

Reply:
> If it is negative, it will return false no matter what.

Yes, and that will take time to debug if they're accidentally passing a negative value. It's better to fail loudly and clearly.

> How about a war[n]ing instead?

A warning might be hard to see in other logging output. An exception will be much clearer and prompt the user to fix their code.